cmd/api-resource-collector/scap.go

```diff
@@ -0,0 +1,235 @@
+	return nil
+}
+
+func parseContent(f *os.File) (*utils.XMLDocument, error) {
+	return utils.ParseContent(bufio.NewReader(f))
+}
+
+// Returns the file, but only after it has been created by the other init container.
+// This avoids a race.
+func openNonEmptyFile(filename string) (*os.File, error) {
+	readFileTimeoutChan := make(chan *os.File, 1)
+
+	// gosec complains that the file is passed through an evironment variable. But
+	// this is not a security issue because none of the files are user-provided
+	cleanFileName := filepath.Clean(filename)
+
+	go func() {
+		for {
+			// Note that we're cleaning the filename path above.
+			// #nosec
+			file, err := os.Open(cleanFileName)
+			if err == nil {
+				fileinfo, err := file.Stat()
+				// Only try to use the file if it already has contents.
+				if err == nil && fileinfo.Size() > 0 {
+					readFileTimeoutChan <- file
+				}
+			} else if !os.IsNotExist(err) {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}()
+
+	select {
+	case file := <-readFileTimeoutChan:
+		fmt.Printf("File '%s' found, using.\n", filename)
+		return file, nil
+	case <-time.After(time.Duration(contentFileTimeout) * time.Second):
+		fmt.Println("Timeout. Aborting.")
+		os.Exit(1)
+	}
+
+	// We shouldn't get here.
+	return nil, nil
+}
+
+func (c *scapContentDataStream) FigureResources() error {
+	paths := getResourcePaths(c.dataStream)
+	if len(paths) == 0 {
+		return fmt.Errorf("no checks found in datastream")
+	}
+	c.resources = paths
+	return nil
+}
+
+// Collect the resource paths for objects that this scan needs to obtain, and their corresponding file paths.
+// Our initial ugly search trail is:
+//  - find "yamlfilecontent_test" nodes. This is all of them (not filtered by profile)
+//  - for each find the test id and jump to the definition
+//  - find "objectpath" and "file" from the check's ref_id
+// This search trail works with
+//
+//        scanType: Platform
+//        profile: xccdf_org.ssgproject.content_profile_platform-moderate
+//        content: ssg-ocp4-ds.xml
+//        contentImage: quay.io/mrogers950/ocp4-openscap-content:platform
+//
+// TODO: The real check takes selected profile into account and goes something like this (rough notes):
+//  - Find "Profile" with id == selected ID.
+//		- if none, error.
+//	- Find each "selected" child.
+//  - If selected=="true" and idref==prefix(xccdf_org.ssgproject.content_rule_)
+//     - strip off xccdf_org.ssgproject.content_rule_ prefix and save that id.
+// 	- use saved ID to find "reference" with ref_id==id
+func getResourcePaths(ds *utils.XMLDocument) map[string]string {
+	out := map[string]string{}
+
+	LOG("Searching for API paths and filenames from datastream..")
+	nodes := ds.Root.Query("//yamlfilecontent_test")
+	for _, node := range nodes {
+		id := node.GetAttributeValue("id")
+		DBG("yamlfilecontent_test id: '%s'", id)
+		if !strings.HasPrefix(id, "oval:ssg-test_") || !strings.HasSuffix(id, ":tst:1") {
```

Comment:
what is `:tst:1`? Might wanna comment the code for that.

Reply:
This will change